### PR TITLE
Use snapshot_pid function instead of snapshot_one in tests

### PR DESCRIPTION
**Pull request by mpozniak95**
## Description
Functions snapshot_one_profile and snapshot_one_collapse are replaced by snapshot_pid_profile and snapshot_one_collapse in test_java.py and test_sanity.py tests.

## Related Issue
https://github.com/Granulate/gprofiler/issues/514

## How Has This Been Tested?
Run test_java and test_sanity

## Checklist:
- [x] I have read the **CONTRIBUTING** document.
- [ ] I have updated the relevant documentation.
- [ ] I have added tests for new logic.

### CODE REVIEW

File: tests/test_java.py [hunk 1]
```diff
@@ -42,8 +42,7 @@
     _application_docker_container,
     assert_function_in_collapsed,
     make_java_profiler,
-    snapshot_one_collapsed,
```
Comment (Jongy) [inline tests/test_java.py:45]: Now that `snapshot_one_*` are not used - let's remove them?

Comment (mpozniak95) [reply tests/test_java.py:45]: Removed in last commit

**Comment by Jongy** [comment]
CI failure seems like somethign was broken in our dependencies and a freshly installed flake8 doesn't run - https://github.com/Granulate/gprofiler/actions/runs/3197282546/jobs/5220663090

We probably need to downgrade the version of flake8 or another library. @mpozniak95 you can check it out, otherwise I will reach it later today.

**Comment by Jongy** [comment]
@mpozniak95 you can merge from master, I solved the flake8 issue.

**Comment by mpozniak95** [comment]
Merged from master

**Comment by Jongy** [comment]
Flaky failure:
```
E       [2022-10-07 08:28:01,861] ERROR: gprofiler.profilers.profiler_base: JavaProfiler: failed to profile process 7 (java)
E       Traceback (most recent call last):
E         File "granulate_utils/linux/ns.py", line 248, in _switch_and_run
E         File "gprofiler/profilers/java.py", line 230, in _run_java_version
E         File "gprofiler/utils/__init__.py", line 249, in run_process
E         File "gprofiler/utils/__init__.py", line 197, in _reap_process
E         File "subprocess.py", line 863, in communicate
E         File "subprocess.py", line 1525, in _communicate
E         File "selectors.py", line 351, in register
E         File "selectors.py", line 237, in register
E         File "selectors.py", line 224, in _fileobj_lookup
E         File "selectors.py", line 39, in _fileobj_to_fd
E       ValueError: Invalid file object: <_io.BufferedReader name=11>
```

Merging this :)

**Review by Jongy — APPROVED** [review]

**Pull request merged**